idl/storage_proxy.idl.hh

```diff
@@ -22,16 +22,22 @@
 #include "idl/keys.idl.hh"
 #include "idl/uuid.idl.hh"
 
-verb [[with_client_info, with_timeout, one_way]] mutation (frozen_mutation fm, inet_address_vector_replica_set forward, gms::inet_address reply_to, unsigned shard, uint64_t response_id, std::optional<tracing::trace_info> trace_info [[version 1.3.0]], db::per_partition_rate_limit::info rate_limit_info [[version 5.1.0]]);
+namespace service {
+    struct fencing_token {
+        int64_t topology_version;
+    };
+}
+
+verb [[with_client_info, with_timeout, one_way]] mutation (frozen_mutation fm, inet_address_vector_replica_set forward, gms::inet_address reply_to, unsigned shard, uint64_t response_id, std::optional<tracing::trace_info> trace_info [[version 1.3.0]], db::per_partition_rate_limit::info rate_limit_info [[version 5.1.0]], std::optional<service::fencing_token> fence [[version 5.2.0]]);
 verb [[with_client_info, one_way]] mutation_done (unsigned shard, uint64_t response_id, db::view::update_backlog backlog [[version 3.1.0]]);
```

Comment:
Why is it optional here?

Reply:
Because during upgrade to a new Scylla version with fencing upgraded replica can receive mutation from a not yet upgraded one, and it needs a was to forward null [here](https://github.com/scylladb/scylladb/pull/12500/files#diff-b973f4e30060a6f4beaf81347658f565fa895998c849d92e80f574a6255bad75R651)